src/sota_tools/server_credentials.cc

```diff
@@ -65,12 +65,13 @@ ServerCredentials::ServerCredentials(const boost::filesystem::path &credentials_
       if (strcmp(filename, "treehub.json") == 0) {
         json_stream = readArchiveFile(a);
         found_config = true;
-      } else if (strcmp(filename, "client.crt") == 0) {
-        client_cert_ = readArchiveFile(a)->str();
-      } else if (strcmp(filename, "client.key") == 0) {
-        client_key_ = readArchiveFile(a)->str();
-      } else if (strcmp(filename, "root.crt") == 0) {
-        root_cert_ = readArchiveFile(a)->str();
+      } else if (strcmp(filename, "client_auth.p12") == 0) {
+        client_p12_ = readArchiveFile(a)->str();
+        method_ = AuthMethod::kTls;
+      } else if (strcmp(filename, "api_gateway.url") == 0) {
+        use_api_gateway = true;
+        ostree_server_ = readArchiveFile(a)->str();
+        boost::trim_if(ostree_server_, boost::is_any_of(" \t\r\n"));
```

Comment:
Design of ServerCredentials class is really confusing as it encapsulates in a single class context of each type of credentials while effectively just a single type context is used within specific class instance.